### PR TITLE
enable status code summary by default

### PR DESCRIPTION
**Pull request by jeremyandrews**
- rename `--status-codes` to `--no-status-codes` (and `GooseDefault::StatusCodes` to `GooseDefault::NoStatusCodes`)
 - collect and summarize status codes by default
 - update the documentation (to review locally, run `mdbook build src/docs/goose-book --open`)
 - while updating the metrics book page (which originally included `--status-codes`) I also added the html report
 - closes https://github.com/tag1consulting/goose/issues/439

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -2378,11 +2378,11 @@ impl GooseConfiguration {
         // Validate `no_metrics`.
         if self.no_metrics {
             // Status codes are not collected if metrics are disabled.
-            if self.status_codes {
+            if self.no_status_codes {
```
Comment (slashrsm) [inline src/config.rs:2381]: Isn't this the other way around - status codes can't be enabled if we don't have metrics but can be disabled in any case?

Comment (jeremyandrews) [reply src/config.rs]: Thanks for catching that! This check is no longer relevant -- if `--no-metrics` is enabled then status codes are disabled automatically (whether or not `--no-status-codes` are enabled).

File: src/config.rs [hunk 2]
```diff
@@ -2157,11 +2157,11 @@ impl GooseConfiguration {
                         .to_string(),
                 });
             // Can't set `status_codes` on Worker.
```
[the remark this answers is not in the document]
Comment (jeremyandrews) [reply src/config.rs]: Thanks! I had missed one other as well, also fixed.

**Comment by slashrsm** [comment]
Awesome work!

**Pull request merged**